python/fledge/plugins/common/shim/south_shim.py

```diff
@@ -68,7 +68,8 @@ def plugin_shutdown(handle):
 
 def plugin_start(handle):
     _LOGGER.info("plugin_start")
-    return _plugin.plugin_start(handle)
+    if hasattr(_plugin, "plugin_start"):
```

Comment:
Hi @MarkRiddoch ,

What if a plugin has both plugin_poll and plugin_start API  methods implemented ? 

or 

what if plugin_start does not exists/misspelled  in async mode plugin;

or 

what if mistakenly wrote start handle/hook in poll mode?

I think the better way is to
1.  See the message from plugin_info.
2.  Decide whether that it is async or poll from plugin info itself.
3. Expect plugin_start if async is present , expect plugin_poll if poll is present.

cc 
@ashish-jabble 
@praveen-garg